test/spec/filters/truncateSpec.js

```diff
@@ -0,0 +1,97 @@
+describe("TruncateFilter", function() {
+
+  var $scope;
+  var $filter;
+
+  beforeEach(inject(function (_$rootScope_, _$filter_) {
+
+    $scope = _$rootScope_;
+    $filter = _$filter_;
+  }));
+
+  beforeEach(function () {
+
+    $scope.params = {
+      str: 'Operation cannot be fulfilled on namespaces\n"ups-broker": The system is ensuring all content is removed\nfrom this namespace. Upon completion, this\nnamespace will automatically be purged by the system.\n',
+      charLimit: 200,
+      useWordBoundary: false,
+      newlineLimit: null
+    };
+  });
+
+  it('should have a minimum of two parameters defined', function() {
+
+    var str = null;
+    var charLimit = null;
+    var useWordBoundary = $scope.params.useWordBoundary;
+    var newlineLimit = $scope.params.newlineLimit;
+    var filteredString = $filter('truncate')(str, charLimit, useWordBoundary, newlineLimit);
+
+    expect(str).toBeFalsy();
+    expect(charLimit).toBeFalsy();
+    expect(useWordBoundary).toEqual(false);
+    expect(newlineLimit).toBeFalsy();
+
+    expect(filteredString).toEqual(null);
+  });
+
+  it('should return a string with a specific length', function() {
+
+    var str = $scope.params.str;
+    var charLimit = $scope.params.charLimit;
+    var useWordBoundary = $scope.params.useWordBoundary;
+    var newlineLimit = $scope.params.newlineLimit;
+    var filteredString = $filter('truncate')(str, charLimit, useWordBoundary, newlineLimit);
+
+    expect(filteredString.length).toEqual(200);
+  });
+
+  it('should return a string that ends on a word boundary', function() {
+
+    var str = $scope.params.str;
+    var charLimit = $scope.params.charLimit;
+    var useWordBoundary = true;
+    var newlineLimit = $scope.params.newlineLimit;
+    var filteredString = $filter('truncate')(str, charLimit, useWordBoundary, newlineLimit);
+
+    expect(filteredString.length).toEqual(188);
```

Comment:
Hm, this looks like a bug? It should not look more than 10 characters back for word boundaries, so I'd expect the smallest length to be 190.